### ISSUE TITLE
Can we re-run tests ?

### ISSUE THREAD

**Issue by slaneyrw** [Modernizr/Modernizr#1911]
It appears that when Modernizr executes it set the state of all tests and it is not possible to re-evaluate any/all tests.

The reason I ask is that we have some conditional polyfills that would change a test from false to true.  Actually working our when the polyfill script is downloaded AND executed is problematic cross-browser so we need constantly check.

At the moment I'm explicitly setting the Modernizr.{test} to true at the end of the polyfill but it would be nice just to ask Modernizr to re-evaluate all loaded tests.

**Comment by patrickkettner** [comment]
No, there is not a way to rerun Modernizr currently. 
Out of curiosity, why are you updating the values? If you are loading a polyfill then it should be trued in all situations, no? What would be the need?

**Comment by slaneyrw** [comment]
I'm using a RequireJS plugin to conditionally load polyfills if modernizr reports a missing feature, and was have issues in some old mobile browsers detecting when the script was actually executed by the browser, not just loaded and parsed.  The script element's "readystatechange" event ( yes it was IE-base engine ) was firing and reporting an appropriate readyState, but the script still hadn't executed.

backstory - my javascript application cannot start until all polyfills loaded and executed as we code for ES6 features.

**Comment by patrickkettner** [comment]
what browser was this?

**Comment by slaneyrw** [comment]
From memory it was a WinPhone 7 or 7.5 IE, but I also iPods with an older version of iOS and safari was problematic as well

**Comment by onassar** [comment]
Jumping in here: I noticed some tests return false positives when run within a iframe that's being hidden. Being able to re-run tests (eg. when the iframe is shown) would be helpful.

**Comment by emados** [comment]
Rerunning tests or being able to invalidate them is also needed for our scenario, switching to / from responsive mode while testing we need the `touchevents` flag to be reset. What's the reason this not implemented so far?

**Comment by Markel** [comment]
> What's the reason this not implemented so far?

I would have to check, but probably v4 allows to re run tests with `testRunner()`, but getting it merged is going to be a great effort #2594

(As a note to myself, I should get on it, at least solving the conflicts which was the main objetive when I took it from #2108)

But yeah, it simply hasn't been a priority 🤷

**Comment by onassar** [comment]
Fair enough. I wonder if it'll become a bigger priority, as it looks like Chrome is changing pretty notably in at least a couple cases whereby things are failing.
